pvlib/pvsystem.py

```diff
@@ -153,11 +158,53 @@ def __init__(self,
         self.racking_model = racking_model
 
     def __repr__(self):
-        attrs = ['name', 'surface_tilt', 'surface_azimuth', 'module',
-                 'inverter', 'albedo', 'racking_model']
+        attrs = ['name', 'model_dc', 'surface_tilt', 'surface_azimuth',
+                 'module', 'inverter', 'albedo', 'racking_model']
         return ('PVSystem: \n  ' + '\n  '.join(
             ('{}: {}'.format(attr, getattr(self, attr)) for attr in attrs)))
 
+    def calc_model_dc(self, irr, temp):
+        """
+        Calculate the DC model
+
+        Use the model specified in model_dc, which may be a user-defined
+        function.
+
+        TODO Details
+        """
+
+        if self.model_dc == 'campanelli':
+            return sdm(get_coeffs_campanelli(irr, temp,
+                       **self.module_parameters),
+                       modules_per_string=self.modules_per_string,
+                       strings_per_inverter=self.strings_per_inverter)
```

Comment:
Hmmm..., I think I should be more consistent here and make `sdm` have the same API/signature as `sapm` and `pvwatts_dc`. This might mean that the `get_coeff_*` call is determined somehow from the `module_parameters` and done inside `sdm`.